test/functional/settings_controller_test.rb

```diff
@@ -27,10 +27,9 @@ class SettingsControllerTest < ActionController::TestCase
 
       context "user has mfa disabled" do
         setup { get :edit }
-        should redirect_to("the setup mfa page") { new_multifactor_auth_path }
-
-        should "set mfa_redirect_uri" do
-          assert_equal edit_settings_path, session[:mfa_redirect_uri]
+        should "flash setup_required_html warning message" do
```

Comment:
Nit: The test description shouldn't be tightly coupled to implementation details. The test description shouldn't change if we change the translation reference. So, if we change the reference from `t("multifactor_auths.setup_required_html")` to `t("multifactor_auths.strong_mfa_level_required_html") for instance, the test description shouldn't need to change, only our assertion does. We can simplify the test description to convey the important part, that a flash message should appear (e.g. `flash a warning message` or something along the lines of this).